AKR:OTR:VKT:YKI:SHARED(Frontend) OPHAKRKEH-515 frontin versiopäivityksiä

jrkkp opened the pull request:
## Yhteenveto

Frontin versionostoja

jrkkp committed:
AKR:OTR:VKT:YKI:SHARED(Frontend) OPHAKRKEH-515 version bumps

jrkkp committed:
AKR:OTR:VKT:YKI:SHARED(Frontend) OPHAKRKEH-515 fix external repo protocol

jrkkp committed:
AKR:OTR:VKT:YKI:SHARED(Frontend) OPHAKRKEH-515 version bumps

jrkkp committed:
AKR:OTR:VKT:YKI:SHARED(Frontend) OPHAKRKEH-515 version bump fixes

jrkkp committed:
AKR:OTR:VKT:YKI:SHARED(Frontend) OPHAKRKEH-515 bump node version

jrkkp committed:
AKR:OTR:VKT:YKI:SHARED(Frontend) OPHAKRKEH-515 version bump fixes

jrkkp committed:
AKR:OTR:VKT:YKI:SHARED(Frontend) OPHAKRKEH-515 jest migration test fixes

jrkkp committed:
AKR:OTR:VKT:YKI:SHARED(Frontend) OPHAKRKEH-515 jest migration test fixes

jrkkp committed:
AKR:OTR:VKT:YKI:SHARED(Frontend) OPHAKRKEH-515 jest migration test fixes

jrkkp committed:
AKR:OTR:VKT:YKI:SHARED(Frontend) bump node versions

jrkkp committed:
AKR:OTR:VKT:YKI:SHARED(Frontend) router fixes

jrkkp committed:
AKR:OTR:VKT:YKI:SHARED(Frontend) skip some cypress tests for now

jrkkp committed:
Merge branch 'dev' into feature/OPHAKRKEH-515

jrkkp committed:
AKR:OTR:VKT:YKI:SHARED(Frontend) linter fixes

jrkkp committed:
AKR:OTR:VKT:YKI:SHARED(Frontend) jest snapshot updates and skip one test for now

jrkkp committed:
AKR:OTR:VKT:YKI:SHARED(Frontend) for some reason local jest test renders invalid dates

jrkkp committed:
AKR:OTR:VKT:YKI:SHARED(Frontend) fix error toast. Also revert date picker back to 5.x

jrkkp committed:
SHARED(Frontend) update shared jest snapshot

pkoivisto commented on frontend/packages/akr/src/tests/cypress/integration/clerk/translatorOverview/clerk_translator_details.spec.ts:
```diff
@@ -266,51 +266,51 @@ describe('ClerkTranslatorOverview:ClerkTranslatorDetails', () => {
   });
 
-  it('should display a confirmation dialog if the back button is clicked and there are unsaved changes', () => {
+  it.skip('should display a confirmation dialog if the back button is clicked and there are unsaved changes', () => {
```
Onko jäänyt ylimääräinen skip vai tarvitaanko tämä edelleen?

pkoivisto commented on frontend/packages/otr/src/tests/cypress/integration/interpreterOverview/clerk_interpreter_details.spec.ts:
```diff
@@ -121,16 +121,16 @@ describe('ClerkInterpreterOverview:ClerkInterpreterDetails', () => {
   });
 
-  it('should display a confirmation dialog if the back button is clicked and there are unsaved changes', () => {
+  it.skip('should display a confirmation dialog if the back button is clicked and there are unsaved changes', () => {
```
Tarvitaanko skip edelleen?

pkoivisto commented on frontend/packages/otr/src/tests/cypress/integration/public_home_page.spec.ts:
```diff
@@ -8,13 +8,13 @@ describe('PublicHomePage', () => {
   });
 
-  it('should allow combining different options to filter interpreters', () => {
+  it.skip('should allow combining different options to filter interpreters', () => {
```
Tarvitaanko skip edelleen?

pkoivisto replied:
Tämä nyt enabloitu, hieman piti muuttaa testissä käytettyä selektoria.

pkoivisto commented on frontend/packages/shared/src/components/CustomDatePicker/CustomDatePicker.tsx:
```diff
@@ -44,14 +40,12 @@ export const CustomDatePicker = ({
     >
       <DesktopDatePicker
         className="custom-date-picker"
-        showToolbar={showToolbar}
```
Onko tämä propsu poistunut kokonaan datepickerin päivityksen yhteydessä?

pkoivisto commented on frontend/packages/shared/src/components/CustomDatePicker/CustomDatePicker.tsx:
```diff
@@ -44,14 +40,12 @@ export const CustomDatePicker = ({
         disabled={disabled}
         minDate={minDate ?? dayjs(MIN_DATE)}
         maxDate={maxDate ?? dayjs(MAX_DATE)}
-        toolbarPlaceholder={placeholder}
```
Onko tämä propsu poistunut kokonaan datepickerin päivityksen yhteydessä?

jrkkp replied:
Joo, datepicker 6 versiossa nämä propsut poistuu. Mutta palautin 5 version niin periaatteessa nämä voisi kyllä palauttaa mutta sitten jos joskus päivitetään niin ne pitäisi poistaa taas. 🤔 En huomannu tosin tällä olevan vaikutusta käytännössä (tai sitten joku datepicker instanssi on menny mun testailuissa ohi).

pkoivisto replied:
Näköjään tätä `placeholder`ia ei codebasessa käytetä, niin eipä taida olla väliä tällä.

pkoivisto commented on frontend/packages/shared/src/components/ScrollToTop/ScrollToTop.tsx:
```diff
@@ -1,18 +1,12 @@
 
-    return unlisten;
-  }, [navigator]);
+  useLayoutEffect(() => {
+    window.scrollTo({ left: 0, top: 0 });
+  }, [location]);
```
Muuttuuko tän ScrollToTopin toiminta jotenkin oleellisesti tämän muutoksen myötä? Jos oikein käsitin, niin erot ovat seuraavat:
- vanha implementaatio skrollasi ylös vanhan sivun unmountin yhteydessä
- ... paitsi jos unmountin aiheutti paluu taaksepäin selaimen historiassa? 
- uusi implementaatio skrollaa ylös uuden sivun mountin yhteydessä

jrkkp replied:
Hyvä huomio! Täytyykin vielä perehtyä tähän vähän paremmin.

jrkkp replied:
Sain nyt ehkä toimimaan oikein

pkoivisto commented on frontend/packages/vkt/src/tests/cypress/integration/clerk_create_exam_event_page.spec.ts:
```diff
@@ -18,12 +18,13 @@ describe('ClerkCreateExamEventPage', () => {
     onClerkExamEventCreatePage.inputMaxParticipants(20);
     onClerkExamEventCreatePage.clickIsHiddenToggle();
-    onClerkExamEventCreatePage.saveButtonEnabledIs(true);
+    // FIXME manual date inputting is broken
+    //onClerkExamEventCreatePage.saveButtonEnabledIs(true);
```
Oliko tän testilogiikan korjaamiseen selvät sävelet?

jrkkp replied:
Hups. Korjasin.

pkoivisto commented on frontend/packages/vkt/src/tests/cypress/integration/examEventOverview/clerk_exam_event_details.spec.ts:
```diff
@@ -129,12 +130,12 @@ describe('ClerkExamEventOverview:ClerkExamEventDetails', () => {
   });
 
-  it('should display a confirmation dialog if the back button is clicked and there are unsaved changes', () => {
+  it.skip('should display a confirmation dialog if the back button is clicked and there are unsaved changes', () => {
```
Skip edelleen tarpeellinen?

pkoivisto commented on frontend/packages/yki/jest-environment-patched-jsdom.js:
```diff
@@ -0,0 +1,13 @@
+// test.environment.js
+
+const JSDOMEnvironment = require("jest-environment-jsdom").default;
+
+module.exports = class CustomTestEnvironment extends JSDOMEnvironment {
+    async setup() {
+        await super.setup();
+        this.global.TextEncoder = TextEncoder;
+        this.global.TextDecoder = TextDecoder;
+        this.global.Response = Response;
+        this.global.Request = Request;
+    }
+};
```
Mihin nämä uudet `jest-environment-patched-jsdom.js`-tiedostot liittyvät?

jrkkp replied:
Tää liittyi siihen, että jsdom päivityksessä nuo globaalit oli poistunu mutta jokin riippuvuus ykissä tarvi niitä vielä. En nyt muista mikä kirjasto se oli. Taisi olla tämä minkä kautta löysin tän ratkaisun: https://stackoverflow.com/questions/68468203/why-am-i-getting-textencoder-is-not-defined-in-jest

Lisäsin myös tähän tiedostoon kommentiksi tämän.

pkoivisto commented on frontend/packages/yki/package.json:
```diff
@@ -22,11 +22,11 @@
     "yki:test:cypress": "TZ=Europe/Helsinki yarn g:cypress run",
     "yki:test:cypress:open": "TZ=Europe/Helsinki yarn g:cypress open",
-    "yki:test:jest": "yarn g:jest",
+    "yki:test:jest": "yarn g:jest --verbose --runInBand --detectOpenHandles --forceExit",
```
Onko tämä muutos jäänne jostain debuggailusta?

jrkkp replied:
Juu, taisi tuosta jsdom debuggailusta jäädä.

jrkkp committed:
VKT:YKI(Frontend) Review fixes

jrkkp committed:
VKT(Frontend) Review fixes

jrkkp committed:
SHARED(Frontend) Review fixes, attempt to fix ScrollToTop

pkoivisto commented on frontend/packages/akr/src/components/clerkTranslator/authorisation/AuthorisationFields.tsx:
```diff
@@ -30,7 +30,7 @@ import { AuthorisationUtils } from 'utils/authorisation';
   authorisation: Authorisation;
   setAuthorisation: (
-    a: ((prevState: Authorisation) => Authorisation) | Authorisation
+    a: ((prevState: Authorisation) => Authorisation) | Authorisation,
```
Näitä loppupilkkuja on tässä PR:ssä lisätty aika runsaasti. Onko nää kenties seurausta jonkun linttaus-depsun päivityksestä tms? Tuottaa aika paljon ylimääräistä hälyä ja voi ehkä joku merkityksellinen muutos jäädä huomaamatta sen seurauksena.

jrkkp replied:
Joo, lintterin päivityksestä. Ehkä defaultit muuttunu lintterissä päivityksen yhteydessä. Katson jos saisin jollain configilla tämän säädettyä samalla tavalla kuin aiemmin.

pkoivisto replied:
Jees. Yksi vaihtoehtohan voisi olla jättää toi lintterin päivitys pois tästä PR:stä ja päivittää se erillisessä PR:ssä tyylimuutoksineen.

pkoivisto replied:
Tai no, ehkä voi jäädä myös tähän nykyiseen kuosiin. Taisin saada selailtua kaikki filut läpi.

pkoivisto replied:
Äh. Kokeilinpa testata näitä muutoksia lokaalisti ja näköjään ainakin mulla prettier valittaa noista ylimääräisistä pilkuista. Ajoin `yarn install` ja sen jälkeen `yarn yki:start:dev-server`. Jäiköhän tästä joku askel nyt tekemättä?

jrkkp replied:
Onko 20.x versio nodesta? Ja ainakin mulla tuntui välillä vaativan, että poistaa node_modules tuolta frontend alta.

pkoivisto replied:
Joo, toi node_modules poistaminen taisi taas kerran olla avain onneen.

pkoivisto commented on frontend/packages/yki/src/tests/jest/components/reassessment/EvaluationOrderPage.test.tsx:
```diff
@@ -28,12 +28,12 @@ describe('EvaluationOrderPage', () => {
   };
 
-  it('should look as expected', () => {
+  it.skip('should look as expected', () => {
```
Tässä vielä yksi uusi testin skippaus.

pkoivisto reviewed (commented):
Näyttää hyvältä ja ainakin YKIn frontti näytti ihan ok edelleen toimivan pikaisen testauksen perusteella. Nämä skipatut testit voisi vielä katsoa kondikseen, muuten mielestäni mergausvalmis.

pkoivisto committed:
YKI(Frontend): Fix test setup: need to use createBrowserRouter(..) with DefaultProviders

jrkkp committed:
AKR:OTR:VKT:YKI:SHARED(Frontend) useBlocker fixes

pkoivisto committed:
OTR(Frontend): Fix and enable failing test.

The old selector didn't for some reason want to find the correct element to type into. We "fix" this by instead using a role query, which is more in line with the testing library matching philosophy.

pkoivisto commented on frontend/packages/shared/src/hooks/useNavigationProtection/useNavigationProtection.ts:
```diff
@@ -1,5 +1,5 @@
 import { useCallback, useEffect } from 'react';
-import { Location, unstable_useBlocker as useBlocker } from 'react-router-dom';
+import { Location, unstable_useBlocker as useBlocker, unstable_usePrompt as usePrompt } from 'react-router-dom';
```
Ylimääräinen import `usePrompt`?

jrkkp committed:
SHARED(Frontend) removed unused import

jrkkp committed:
Merge branch 'dev' into feature/OPHAKRKEH-515
